src/cairoWriter.ts

```diff
@@ -716,7 +717,7 @@ class FunctionCallWriter extends CairoASTNodeWriter {
           }
         } else if (
           node.vReferencedDeclaration instanceof CairoFunctionDefinition &&
-          node.vReferencedDeclaration.functionStubKind === FunctionStubKind.StructDefStub
+          node.vReferencedDeclaration.acceptsRawDarray
         ) {
           return [`${func}(${args}_len, ${args})`];
         }
```

Comment:
I can't comment on the specific line, but line 734 checks address literals.
The starknet field prime is slightly bigger than 2 ** 251.
FIELD_PRIME = 2 ** 251 + 17 * 2 ** 192 + 1
[starkware-libs/cairo-lang@cf8266f/src/starkware/crypto/starkware/crypto/signature/nothing_up_my_sleeve_gen.py#L35](https://github.com/starkware-libs/cairo-lang/blob/cf8266fd5d1ff66962579ff7967ac5cdcf699f77/src/starkware/crypto/starkware/crypto/signature/nothing_up_my_sleeve_gen.py#L35)

Reply:
We asked at a recent meeting with Starkware and they said the addresses are 251 bits wide. Is there something that suggests they take up the fult set of felt values?

Reply:
Nope, I just assumed that they would take the full felt space.